Demo/Demo/SwiftUIComponents/CardVaultViews/CardVaultView.swift

```diff
@@ -0,0 +1,71 @@
+import SwiftUI
+
+struct CardVaultView: View {
+
+    @StateObject var baseViewModel = BaseViewModel()
+    @StateObject var cardVaultViewModel = CardVaultViewModel()
+
+    private let cardFormatter = CardFormatter()
+
+    // MARK: Views
+
+    var body: some View {
+        ScrollView {
+            ScrollViewReader { scrollView in
+                VStack(spacing: 16) {
+                    CreateSetupTokenView(baseViewModel: baseViewModel, cardVaultViewModel: cardVaultViewModel)
```

Comment:
nit: There may be a way to provide an `onSubmit` action to `CreateSetupTokenView` as an `@escaping` closure. Then we could remove the view model parameters from the constructor.

Reply:
Similar to [this post](https://developer.apple.com/forums/thread/718378). I'm not sure if this is kosher for SwiftUI, but it's a common delegation pattern we use in Compose for Android.

Reply:
Yes, we can do that but I still need the CardVaultViewModel to get the loading state (which I plan to move to viewModel) or error state from the CreateSetupTokenViews and other views as well.

Reply:
Nice yeah I'm sure it'll be a mixture of different stuff. I saw SwiftUI has some concept of an environment object too that looks cool.

Reply:
Yeah, they're really easy to use but probably use with caution for work. I should just probably pass in merchant integration instead of reference to baseVM into CreateSetupTokenView though.